storage/src/main/java/tech/pegasys/teku/storage/server/pruner/BlobSidecarPruner.java

```diff
@@ -164,13 +165,14 @@ private void pruneBlobsPriorToAvailabilityWindow() {
       if (storeNonCanonicalBlobSidecars) {
         final long nonCanonicalBlobsPruningStart = System.currentTimeMillis();
         final boolean nonCanonicalBlobsLimitReached =
-            database.pruneOldestNonCanonicalBlobSidecars(latestPrunableSlot, pruneLimit);
+            database.pruneOldestNonCanonicalBlobSidecars(
+                latestPrunableSlot, pruneLimit, archiveWriter);
         logPruningResult(
             "Non canonical Blobs pruning finished in {} ms. Limit reached: {}",
             nonCanonicalBlobsPruningStart,
             nonCanonicalBlobsLimitReached);
       }
-    } catch (ShuttingDownException | RejectedExecutionException ex) {
+    } catch (ShuttingDownException | RejectedExecutionException | IOException ex) {
```

Comment:
IOException -> shutting down?

Reply:
Nice catch. separated out the catch and put it as an error.